src/collapse-perf.rs

```diff
@@ -325,23 +325,34 @@ impl PerfState {
 // massage function name to be nicer
 // NOTE: ignoring https://github.com/jvm-profiling-tools/perf-map-agent/pull/35
 fn with_module_fallback(module: &str, rawfunc: &str, pc: &str, include_addrs: bool) -> String {
-    if rawfunc == "[unknown]" {
-        // try to use part of module name as function if unknown
-        let rawfunc = if module != "[unknown]" {
-            // use everything following last / of module as function name
-            &module[module.rfind('/').map(|i| i + 1).unwrap_or(0)..]
-        } else {
-            "unknown"
-        };
+    if rawfunc != "[unknown]" {
+        return rawfunc.to_string()
+    }
 
-        if include_addrs {
-            format!("[{} <{}>]", rawfunc, pc)
-        } else {
-            format!("[{}]", rawfunc)
-        }
+    // try to use part of module name as function if unknown
+    let rawfunc = if module != "[unknown]" {
+        // use everything following last / of module as function name
+        &module[module.rfind('/').map(|i| i + 1).unwrap_or(0)..]
+    } else {
+        "unknown"
+    };
```

Comment:
How about:
How about, above:
```
if module != "[unknown]" {
    // ...
} else if !include_addrs {
    return rawfunc.to_string();
} else {
    "unknown"
}
```
to save us the string construction below for the `rawfunc == module == "[unknown]"` and `!include_addrs` common-case? Might not matter in terms of performance though...

Reply:
Good idea! I might want to write it like this, though :)

```rust
let rawfunc = match (module, include_addrs) {
    ("[unknown]", true) => "unknown",
    ("[unknown]", false) => {
        // no need to process this further
        return rawfunc.to_string();
    }
    (module, _) => {
        // use everything following last / of module as function name
        &module[module.rfind('/').map(|i| i + 1).unwrap_or(0)..]
    }
};
```

Reply:
I like that! Either order is fine I think.